### PR TITLE
CDAP-5029 Changes to show logs on UI on mapreduce failure

### PR DESCRIPTION
**Pull request by CuriousVini**
Changes to show MapReduce logs on UI

Issue: https://issues.cask.co/browse/CDAP-5029
Build: http://builds.cask.co/browse/CDAP-DUT3706-3

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/read/AvroFileReader.java [hunk 1]
```diff
@@ -44,6 +44,7 @@
 public class AvroFileReader {
   private static final Logger LOG = LoggerFactory.getLogger(AvroFileReader.class);
   private static final long DEFAULT_SKIP_LEN = 50 * 1024;
+  public static final int AVRO_SYNC_SIZE = 16;
```
Comment (gokulavasan) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/read/AvroFileReader.java]: Is there a Avro constant that we can use instead?

Comment (CuriousVini) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/read/AvroFileReader.java]: Yes. There is DataFileConstants.SYNC_SIZE available. I will add that.

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogHandler.java [hunk 2]
```diff
@@ -344,9 +344,10 @@ public long getToMillis() {
   }
 
   private static TimeRange parseTime(long fromTimeSecsParam, long toTimeSecsParam, HttpResponder responder) {
+    long currentTimeMillis =  System.currentTimeMillis();
```
Comment (anwar6953) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogHandler.java]: nit: extra space after `=`

Comment (anwar6953) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogHandler.java]: but +1 on the change here

Comment (CuriousVini) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogHandler.java]: Thanks Ali. will address that..

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/read/AvroFileReader.java [hunk 3]
```diff
@@ -181,9 +172,36 @@ public void readLog(Location file, Filter logFilter, long fromTimeMs, long toTim
     }
   }
 
+  private int readToNextSync(Filter logFilter, long fromTimeMs, DataFileReader<GenericRecord> dataFileReader,
+                             List<List<LogEvent>> logSegments, int count, List<LogEvent> logSegment,
+                             long endPosition) throws IOException {
+    GenericRecord datum;
+    // Read till the end if stopPoint = -1 or read until stopPoint has reached
+    while (dataFileReader.hasNext() && (endPosition == -1 || !dataFileReader.pastSync(endPosition))) {
+      datum = dataFileReader.next();
```
Comment (chtyim) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/read/AvroFileReader.java]: It's better to reuse the `GenericRecord`:

```java
GenericRecord datum = null;
while (...) {
  datum = dataFileReader.next(datum);
  ...
}
```

File: cdap-common/src/main/java/org/apache/twill/filesystem/FileContextLocation.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import javax.annotation.Nullable;
+
+/**
+ * Duplicate of twill {@link FileContextLocation} with overridden toString(). Workaround for TWILL-167
+ */
+final class FileContextLocation implements Location {
```
Comment (sreevatsanraman) [inline cdap-common/src/main/java/org/apache/twill/filesystem/FileContextLocation.java]: Is there a JIRA to remove this after a twill release. Can you add that as TODO with JIRA number here?

**Comment by CuriousVini** [comment]
Closing. Created new PR https://github.com/caskdata/cdap/pull/5334

**Pull request closed without merging**